apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -12,21 +7,15 @@ import { ListItemComponent } from '../list-item/list-item.component';
     <div
       class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
       [class]="customClass">
-      <img
-        *ngIf="type === CardType.TEACHER"
-        src="assets/img/teacher.png"
-        width="200px" />
-      <img
-        *ngIf="type === CardType.STUDENT"
-        src="assets/img/student.webp"
-        width="200px" />
+      <ng-content select="[image-header]"></ng-content>
 
       <section>
-        <app-list-item
-          *ngFor="let item of list"
-          [name]="item.firstName"
-          [id]="item.id"
-          [type]="type"></app-list-item>
+        @for (item of list; track item.id) {
+          <app-list-item
+            [name]="item[nameKey]"
+            [id]="item.id"
+            (deleteItem)="onDeleteItem($event)"></app-list-item>
+        }
```

Comment:
this works but it's not very customizable. What will you do if i need to add an icon but only for city Card? and add another button for teacher. you will start to add multiple conditions.
Moreover you will have to modify this component at each new change

Look at `ngTemplateOutlet`